### PR TITLE
Expand the add to cart form test coverage

### PR DESCRIPTION
**Pull request by mglaman**
WIP to improve tests for add to cart form and attribute selection.

### CODE REVIEW

File: modules/cart/src/Tests/AddToCartFormTest.php [hunk 1]
```diff
@@ -288,10 +271,32 @@ public function testExposedLineItemFields() {
...
+   *   The variation type.
+   * @param string $name
+   *   The attribute field name.
+   * @param array $options
+   *   Associative array of key name values. [red => Red].
+   *
+   * @return \Drupal\Core\Entity\EntityInterface[]
+   *   Array of attribute entities.
+   */
+  protected function createAttributeSet(ProductVariationTypeInterface $variation_type, $name, array $options) {
```
Comment (mglaman) [inline modules/cart/src/Tests/AddToCartFormTest.php:425]: Should this and some of its helpers be traits for other product tests, @bojanz? Could put into its own PR

File: modules/cart/src/Tests/AddToCartFormTest.php [hunk 2]
```diff
@@ -146,128 +147,238 @@ public function testMultipleVariationsAddToCartForm() {
...
+    $this->drupalPostAjaxForm(NULL, [
+      'purchased_entity[0][attributes][test_size_attribute]' => $size_attributes['7']->id(),
+    ], 'purchased_entity[0][attributes][test_size_attribute]');
+    $this->assertAttributeExists('edit-purchased-entity-0-attributes-test-color-attribute', $color_attributes['red']->id());
+    $this->assertAttributeExists('edit-purchased-entity-0-attributes-test-color-attribute', $color_attributes['green']->id());
+
+    $this->drupalPostAjaxForm(NULL, [
+      'purchased_entity[0][attributes][test_color_attribute]' => $color_attributes['green']->id(),
+    ], 'purchased_entity[0][attributes][test_color_attribute]');
+    $this->assertAttributeDoesNotExist('edit-purchased-entity-0-attributes-test-size-attribute', $size_attributes['6']->id());
```
Comment (mglaman) [inline modules/cart/src/Tests/AddToCartFormTest.php]: @bojanz this fails. Should it? `6in` is technically not a valid attribute value when checking `Green` product variations. Or should it, because we're exploring all options.

File: modules/cart/src/Tests/AddToCartFormTest.php (hunk 1 above)
Comment (bojanz) [inline modules/cart/src/Tests/AddToCartFormTest.php:425]: Doesn't hurt to have those helpers in the product test base.

**Comment by mglaman** [comment]
@bojanz waited to merge; want your feedback on the commit message.

**Comment by bojanz** [comment]
Committed in 42097368a86763a0e968b3637c35608b8d55a631.
